rero_ils/modules/holdings/api.py

```diff
@@ -98,22 +98,31 @@ def extended_validation(self, **kwargs):
 
         Ensures that holdings of type serials are created only on
         journal documents i.e. serial mode_of_issuance documents.
+
+        Ensures that holdings of type electronic are created only on
+        ebooks documents i.e. harvested documents.
         """
-        document = Document.get_record_by_pid(self.document_pid).dumps()
-        is_serial = self.get('holdings_type') == 'serial'
-        is_issuance = document.get('issuance') == 'rdami:1003'
-        if (is_issuance and not is_serial) or (not is_issuance and is_serial):
-            return False
-        return True
+        document = Document.get_record_by_pid(self.document_pid)
+        is_serial = self.holdings_type == 'serial'
+        is_electronic = self.holdings_type == 'electronic'
+        is_issuance = document.dumps().get('issuance') == 'rdami:1003'
+        return not(
```

Comment:
probably better to write:
```python
return (is_issuance and is_serial) ^ (document.harvested and is_electronic)
```

Reply:
Did not work! can you take a deeper look because now I am worried that my logic is not correct.

Reply:
You can leave at it is.